app/models/concerns/file_backed_model.rb

```diff
@@ -51,8 +51,10 @@ def from_file(filename)
     # Returns a Pathname to location configured in the database through the
     # :setting_name setting
     def pwd
-      conf = Configuration.find_or_create_by(name: @setting_name, value: @default_path)
-      Pathname.new(conf.value)
+      @pwd ||= begin
+        conf = Configuration.find_or_create_by(name: @setting_name, value: @default_path)
```

Comment:
does this work, don't you need for this to be `create_with`, also, shouldn't entire thing be in `Configuration`?